test/lib/absinthe/relay/node/parse_ids_test.exs

```diff
@@ -148,6 +148,7 @@ defmodule Absinthe.Relay.Node.ParseIDsTest do
         end
 
         middleware Absinthe.Relay.Node.ParseIDs,
+          mutation_root: true,
```

Comment:
Can you elaborate on the purpose of this?

Reply:
Whether or not to look for `:input` is never ambiguous from Absinthe.Relay's perspective, I think we should avoid requiring that the client tell us.

Reply:
Yeah. I was going to ask you what you thought of this version. It was one of the ideas I had on how to address this. 

The problem is that `__parse_ids_root` is set at the global resolution level for a mutation using relay. The trouble happens when you try and use `ParseIDs` from a child field inside a mutation query because that `ParseIDs` `__parse_ids_root` is still set for the child's args.

You could have this as an option to pass in from the `mutation` level so `ParseIDs` knows that it actually came from the mutation root instead of a child field.

Reply:
That way, the internals are less fragile as far as ordering and pattern-matching goes.

Reply:
As is though, this wouldn't be backwards compatible, so that would need address somehow.